observation-service/src/main/java/gov/cdc/etldatapipeline/observation/util/ProcessObservationDataUtil.java

```diff
@@ -323,10 +323,12 @@ private void transformObservationCoded(String observationCoded) {
         try {
             JsonNode observationCodedJsonArray = parseJsonArray(observationCoded);
 
+            ObservationCodedKey codedKey = new ObservationCodedKey();
             for (JsonNode jsonNode : observationCodedJsonArray) {
                 ObservationCoded coded = objectMapper.treeToValue(jsonNode, ObservationCoded.class);
-                observationKey.setObservationUid(coded.getObservationUid());
-                sendToKafka(observationKey, coded, codedTopicName, coded.getObservationUid(), "Observation Coded data (uid={}) sent to {}");
+                codedKey.setObservationUid(coded.getObservationUid());
+                codedKey.setOvcCode(coded.getOvcCode());
```

Comment:
Here with Lombok AllArgsConstructor we can simplify the object creation.

Reply:
> Here with Lombok AllArgsConstructor we can simplify the object creation.

Not too good in this case. The key is created once, before the loop, and then being changed in the loop, obtaining new ovcCode on each iteration. It would be not so efficient to create a new instance on each iteration.